### DISCUSSION TITLE
`object-shorthand` but for destructuring assignment?

### DISCUSSION THREAD

**Discussion by ryanrhee** [eslint/eslint#16110] [Q&A] [answered]
```
const foo = {a: "foo"}
// ...
const {a: a} = foo
```

Can I configure eslint to error on the last line? bonus pts if it can autofix to `const {a} = foo`.

EDIT: forgot to mention, I looked at `object-shorthand` which only seems to be for literals, and `prefer-destructuring` which doesn't seem to have any options with respect to declaration shorthands.

**Comment by mdjermanovic** [comment, marked as answer]
> ```
> const foo = {a: "foo"}
> // ...
> const {a: a} = foo
> ```
> 
> 
>     
>   
> 
> Can I configure eslint to error on the last line? bonus pts if it can autofix to `const {a} = foo`

You can use [no-useless-rename](https://eslint.org/docs/latest/rules/no-useless-rename) rule, and it's auto-fixable.

[Playground example](https://eslint.org/play#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVzZWxlc3MtcmVuYW1lOiBbXCJlcnJvclwiXSAqL1xuXG5jb25zdCB7YTogYX0gPSBmb28iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==)

**Reply by ryanrhee** [reply]
thank you!
